internal/controllers/user_devices_controller.go

```diff
@@ -1587,6 +1587,11 @@ func (udc *UserDevicesController) UpdateNFTImage(c *fiber.Ctx) error {
 		return fiber.NewError(fiber.StatusBadRequest, "Vehicle not minted.")
 	}
 
+	// This would only happen if it's a "permissionless mint".
+	if !userDevice.MintRequestID.Valid {
```

Comment:
I named the images in S3 using the meta-transaction id. We can't rely on this existing anymore, so have to bail here. Yev has some plan for getting this all on-chain and in IPFS.